File: Parrot.java
import java.util.ArrayList;
import java.util.Random;

public class Parrot {
    private final ArrayList<String> phrases = new ArrayList<String>();
    private final Random rng = new Random();

    public Parrot(String s) {
        this.addPhrase(s);
    }
    public void say() {
        // Select a random phrase.
        String phrase = this.phrases.get(this.rng.nextInt(0, this.phrases.size()));
        System.out.println(phrase);
    }
    public void say(int n) { // Overloaded method to allow for repetition
        while (n-- > 0) { this.say(); }
    }
    public void addPhrase(String p) {
        this.phrases.add(p);
    }
    public void modifyPhrase(String _old, String _new) {
        int i = this.phrases.indexOf(_old);
        if (i >= 0) {
            this.phrases.set(i, _new);
        }
    }

    public static void main(String[] args) {
        // Begin the ritual
        Parrot p = new Parrot(" ුම් ි ම ීත");
        p.say();
        p.modifyPhrase(" ුම් ි ම ීත", "නෝර්වීජියානු නිල්");
        p.say();
        p.addPhrase("ක ේ ධි දක ද ්මැ ණ් ු ක ත්ප නි ැර න ආක");
        p.addPhrase("ම දාවලවඉසය ුවේ");
        p.addPhrase("ය උ  න සි  නඉධේ න කත්ණාරජ   ල  නසපවඉ");
        p.addPhrase("යජ බවම ජ ානණ ී මඅ න  කහ ව ඉඉම තා ුි ීන");
        p.addPhrase("ය සි ඉ න කතණාරජ ල නසපව");
        p.addPhrase("න   න ණ ේ  තයාදණ් රරරරරරරරරරරරරරරරරරරරරරරරරරරරරරරරර ලසයම ඉ ම ඉ ම ඉ ම ඉ ම ඉ ම ඉ ම ඉ ම ඉ ධ  ඉ ව   ව ඉධරර ම් වව ර ස නම ");
        p.addPhrase("ක ේ ධිදකක ේධ  ේ ධි දක ද ්මැ ණ් ු කත්ප නි ැර න ආකක ද ්මැණ ුකත්පන ැර න");
        p.say(5);
    }
};

File: parrot.cpp
#include <ctime>
#include <string>
#include <vector>

using namespace std;

class Parrot {
	public:
		Parrot(string s) {
			add_phrase(s);
		}
		void say() {
			// Select a random phrase.
			const char* phrase = phrases[rand() % phrases.size()].c_str();
			printf("%s\n", phrase);
		}
		void say(unsigned int n) { // Overloaded method to allow for repetition
			while (n--) { say(); }
		}
		void add_phrase(string p) {
			phrases.push_back(p);
		}
		void modify_phrase(string _old, string _new) {
			for (unsigned int i = 0; i < phrases.size(); i++) {
				if (phrases[i] == _old) { phrases[i] = _new; return; }
			}
		}
	private:
		vector<string> phrases;
};

int main() {
	// Seed the RNG
	srand(time(NULL));
	// Recite the forbidden rites
	Parrot p("ᛢᛗᚯ ᛕᚷ ᛰᛇ᛫");
	p.say();
	p.modify_phrase("ᛢᛗᚯ ᛕᚷ ᛰᛇ᛫", "ᛒᛓᚯ ᚴᚸᛟᚿᚼᚳᚧ ᛨᚤᚸᛩᛅᚱ ᚤᚢᛟᚸᛑ᛫");
	p.say();
	p.add_phrase("ᛘᛕᛣ ᛛᚦᛧᚫᚯᛚ ᛑᚤᛝᛏᛃᚣ ᛟᛢ᛫");
	p.add_phrase("ᚵ ᛩᛠᚪ ᛃᚯᚽᚿᚤᛮ ᚧᛣᚤᚠ ᛇᛎᛃ᛭");
	p.add_phrase("ᛞᚬ ᛀᛗᚰᚲ ᚾᚺᚾᚪᚬ ᛉᛢᛮ᛫");
	p.add_phrase("ᛎᚧᚠᛡᛇᚺ ᚺᛒᚣᛛᚻᚩ᛭");
	p.add_phrase("ᛂᚥᚷ ᛣᛍᛦᛂ ᚳᛛ᛫");
	p.add_phrase("ᛟᛛᛑᛖᛛ ᛛᛐᛄ᛬");
	p.add_phrase("ᛕᛅ ᛢᛛᛄᛆᛄᚩ᛫ ᛐᛞᚬ ᛮᚽ᛬");
	p.say(1);
    return 0;
}

